vktrace/vktrace_layer/vktrace_lib_trace.cpp

```diff
@@ -396,18 +423,18 @@ VKTRACER_EXPORT VKAPI_ATTR VkResult VKAPI_CALL __HOOKED_vkMapMemory(
     VkFlags flags,
     void** ppData)
 {
-    vktrace_trace_packet_header* pHeader;
     VkResult result;
+    vktrace_trace_packet_header* pHeader;
     packet_vkMapMemory* pPacket = NULL;
     VKAllocInfo *entry;
     CREATE_TRACE_PACKET(vkMapMemory, sizeof(void*));
     result = mdd(device)->devTable.MapMemory(device, memory, offset, size, flags, ppData);
     vktrace_set_packet_entrypoint_end_time(pHeader);
     entry = find_mem_info_entry(memory);
 
-    // For vktrace usage, clamp the memory size to the total size less offset if VK_WHOLE_SIZE is specified.
+    // For vktrace usage, clamp the memory size to the total size if VK_WHOLE_SIZE is specified.
     if (size == VK_WHOLE_SIZE) {
-        size = entry->totalSize - offset;
+        size = entry->totalSize;
```

Comment:
Agreed, now that I know that "VK_WHOLE_SIZE" is only for the remainder of the buffer size.